core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -334,26 +294,22 @@ sealed trait Chunk[+A] { self =>
    * Statefully maps over the chunk, producing new elements of type `B`.
    */
   final def mapAccum[S1, B](s1: S1)(f1: (S1, A) => (S1, B)): (S1, Chunk[B]) = {
-    var s: S1          = s1
-    var i              = 0
-    var dest: Array[B] = null.asInstanceOf[Array[B]]
-    val len            = self.length
+    var s    = s1
+    var dest = null.asInstanceOf[Array[B]]
 
-    while (i < len) {
-      val a = self(i)
+    var i = 0
```

Comment:
@vasilmkd The problem with this approach is that capturing a `var` in a closure causes it to be boxed inside an `ObjectRef`. That slows down the loops considerably. I am not sure this tradeoff is worth it for partially avoiding boxing in some cases.

Reply:
Do you think you could benchmark the affect of this change?

Reply:
Learning things the hard way. 😄 Worth a try. Feel free to close the PR.